Axinom/mosaic-media-template
# [AB#39600] [AB#39414] [AB#39279] [AB#39062] Movies and movie genres data can now be synced with the Localization service

# Pull Request

## Prerequisites

- [x] My code follows the coding conventions

## Description

Genral description:
-

- Integration with the Localization service is added for Movies and Movie Genres.
- By enabling the Logical Replication on the media Service database - it is now possible to listen to changes associated with localizable tables and send out messages to the Localization Service.
	- Check `infra/docker-compose.yml` for changes that are required to be done on the Postgres Server level to enable Logical Replication
	- Owner user also must be altered to include Replication (e.g. see `scripts/helpers/db-command-helpers.ts` for sample script.
- To account for existing movies and movie genres - a one-time script is added to sync existing movies with the localization service `src/one-time-register-movie-localization-sources.ts`.
    - It's part of the running service and must be launched from within the deployment environment (e.g. using `node dist/one-time-register-movie-localization-sources`, which assumes that process.env already has config values loaded)
- A dedicated `IS_LOCALIZATION_ENABLED` config variable is added to enable/disable localization functionality in the service (does not affect the database migrations/setup, but affects the data syncing between media and localization services)
- service account token is now cached in memory (to avoid spamming its retrieval, e.g. in case of syncing movies data with localization service during ingest)
- Movie and Movie Genre localizable entity definitions are now synced on media service startup.
- Media Workflows adjusted to connect movie genres and movie details with localization service workflows.

Testing notes:
- 

- Following actions trigger messages to the localization service, the results of which can be observed in localization stations. Make sure that at least one non-default locale is created for the environment for changes to be observable.
	- Create a movie - movie localization created
	- Update movie title and/or description and/or synopsis - movie localization updated
	- Assign/Change/Unassign movie cover (not teaser) - movie localization updated
	- Delete movie - movie localization deleted
	- Create a movie genre -  movie genre localization created
	- Update movie genre title - movie genre localization updated
	- Delete movie genre - movie genre localization deleted
- Ingest of movies must be tested as well
